### PR TITLE
SNOW-1669082: Add support for `Resampler.asfreq`

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
SNOW-1669082

This PR implements `Resampler.asfreq` which follows the same underlying code path as `Series.asfreq` and `DataFrame.asfreq` which was implemented in a previous PR.

### CODE REVIEW

**Review by sfc-gh-rdurrani — APPROVED** [review]
LGTM!

File: src/snowflake/snowpark/modin/pandas/resample.py [hunk 1]
```diff
@@ -280,8 +280,24 @@ def fillna(self, method: str, limit: Optional[int] = None):
...
-    def asfreq(self, fill_value: Optional[Any] = None):  # pragma: no cover
-        self._method_not_implemented("asfreq")
+    def asfreq(self, fill_value: Optional[Any] = None):
+        is_series = not self._dataframe._is_dataframe
+
+        if fill_value is not None:
+            # TODO: SNOW-1660802: Implement `fill_value` parameter once `GroupBy.fillna` is supported
+            ErrorMessage.not_implemented(
+                "Parameter fill_value of resample.asfreq has not been implemented."
+            )
```
Comment (sfc-gh-joshi) [inline src/snowflake/snowpark/modin/pandas/resample.py:290]:
```suggestion
            ErrorMessage.parameter_not_implemented_error("fill_value", "Resample.asfreq")
```

**Review by sfc-gh-joshi — APPROVED** [review]

File: tests/integ/modin/resample/test_resample_asfreq.py [hunk 2]
```diff
@@ -56,3 +69,5 @@ def test_asfreq_negative():
         snow_df.asfreq(freq="5s", method="ffill", normalize=True)
     with pytest.raises(NotImplementedError):
         snow_df.asfreq(freq="5s", method="ffill", fill_value=2)
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/modin/resample/test_resample_asfreq.py:71]: Why are we testing `DataFrame.asfreq` here when the test file is `test_resample_asfreq`?

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM overall. Just a small question.

File: tests/integ/modin/resample/test_resample_asfreq.py (hunk 2 above)
Comment (sfc-gh-nkrishna) [reply tests/integ/modin/resample/test_resample_asfreq.py:71]: `DataFrame/Series.asfreq` end up just calling the `resample` implementation. I didn't realize at that time the `Resampler` class also has an `asfreq` so this PR just adds that. That's why all 3 are tested in the same file under the resample folder.

**Pull request merged**